msb-compat/src/test/scala/DecoderConverterSpec.scala

```diff
@@ -596,8 +596,8 @@ private[reactivemongo] trait DecoderConverterSpec {
         val codec: Codec[BSONMaxKey] = handler
         val dec: Decoder[BSONMaxKey] = reader
 
-        decode(new BsonMaxKey, dec) must beSuccessfulTry(BSONMaxKey) and {
```

Comment:
Unrelated changed, should be reverted for me

Reply:
Oh yeah. The darn auto-formatting... I'll fix that.

Reply:
Oh, that's not the auto-formatting. The change was necessary, otherwise it would not compile...

Reply:
```
[error] /msb-compat/src/test/scala/DecoderConverterSpec.scala:599:57: type mismatch;
[error]  found   : org.specs2.matcher.TrySuccessCheckedMatcher[reactivemongo.api.bson.BSONMaxKey.type]
[error]  required: org.specs2.matcher.Matcher[scala.util.Try[reactivemongo.api.bson.BSONMaxKey]]
[error]         decode(new BsonMaxKey, dec) must beSuccessfulTry(BSONMaxKey) and {
[error]                                                         ^
[error] /msb-compat/src/test/scala/DecoderConverterSpec.scala:600:61: type mismatch;
[error]  found   : org.specs2.matcher.TrySuccessCheckedMatcher[reactivemongo.api.bson.BSONMaxKey.type]
[error]  required: org.specs2.matcher.Matcher[scala.util.Try[reactivemongo.api.bson.BSONMaxKey]]
[error]           decode(new BsonMaxKey, codec) must beSuccessfulTry(BSONMaxKey)
[error]                                                             ^
[error] two errors found
```

Reply:
Then there is a issue, as it was working.

Reply:
That only applies to Scala 2.13 though, which was not working at all before...

Reply:
These tests are not only for 2.13. Anyway need to have a look at that.

Reply:
Seems like the Scala compiler < 2.13 (*helpfully* ?) infers the wrong type, which is why this worked before. Scala 2.13 is pickier here, which is also correct, IMHO.

With Scala 2.12 you could construct something broken like this:
```
def f: Try[BsonMaxKey] = Success(new BsonMaxKey)

f must be beSuccessfulTry(BsonMaxKey)
```
This actually compiles but throws an error at runtime:
```
Success(...) is a Success but BsonMaxKey@6d6ed90c != BsonMaxKey$@2753024
```
What do you think?